meta-lmp-bsp/dynamic-layers/stm-st-stm32mp/recipes-support/flashlayouts-stm32mp1/flashlayouts-stm32mp1.bb

```diff
@@ -6,14 +6,48 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MPL-2.0;md5=815ca599c9df247a0c7
 
 inherit deploy
 
-SRC_URI = "file://FlashLayout_emmc_stm32mp157a-ev1-trusted.tsv.in"
+DEPENDS = "sdcard-raw-tools-native"
+
+MFGTOOL_FLASH_IMAGE ?= "lmp-base-console-image"
+
+LMP_FLASHLAYOUT_TEMPLATE ?= "FlashLayout_stm32mp1-trusted.tsv.in"
+LMP_FLASHLAYOUT_IMAGE = "${MFGTOOL_FLASH_IMAGE}-${MACHINE}.${@bb.utils.contains('DISTRO_FEATURES', 'sota', 'ota-ext4', 'ext4', d)}"
+
+SRC_URI = "file://${LMP_FLASHLAYOUT_TEMPLATE}"
 
 do_compile() {
-    sed -e 's/@@IMAGE@@/${IMAGE}/' ${WORKDIR}/${LMP_FLASHLAYOUT}.in > ${WORKDIR}/${LMP_FLASHLAYOUT}
+    sed -e 's/@@IMAGE@@/${MFGTOOL_FLASH_IMAGE}/' \
+        -e 's/@@MACHINE@@/${MACHINE}/' \
+        -e 's/@@BOARD_NAME@@/${LMP_FLASHLAYOUT_BOARD_NAME}/' \
+        -e 's/@@BOARD_DISK@@/${LMP_FLASHLAYOUT_BOARD_DISK}/' \
+        -e 's/@@BOARD_OFFSET_FSBL1@@/${LMP_FLASHLAYOUT_BOARD_OFFSET_FSBL1}/' \
+        -e 's/@@BOARD_OFFSET_FSBL2@@/${LMP_FLASHLAYOUT_BOARD_OFFSET_FSBL2}/' \
+        -e 's/@@BOARD_OFFSET_SSBL@@/${LMP_FLASHLAYOUT_BOARD_OFFSET_SSBL}/' \
+        -e 's/@@BOARD_OFFSET_ROOT@@/${LMP_FLASHLAYOUT_BOARD_OFFSET_ROOT}/' \
+        ${WORKDIR}/${LMP_FLASHLAYOUT_TEMPLATE} > ${WORKDIR}/${LMP_FLASHLAYOUT}
 }
 
-do_deploy() {
-    install -m 0644 ${WORKDIR}/${LMP_FLASHLAYOUT} ${DEPLOYDIR}
+# deploy a .tar.gz file of artifacts
+do_deploy_append() {
```

Comment:
No need to append if this is the only definition you have. Just do_deploy (as it was previously) should be fine.